File: leetcode2/0006. ZigZag Conversion.py
class Solution(object):
    def convert(self, s,numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows==1:
            return s
        
        rep=['']*numRows
        k=0
        while k<len(s):
            if (k%(numRows+numRows-2))<numRows:
                rep[k%(numRows+numRows-2)]+=s[k]
                k+=1
            else:# (k%(numRows+numRows-2))<numRows+numRows-2:
                rep[numRows-2-k%(numRows+numRows-2)]+=s[k]
                k+=1
        res=''
        for s in rep:
            res+=s
        return res
                

File: archives/C#/0006. ZigZag Conversion.cs
public class Solution {
    public string Convert(string s, int numRows) {
        if(s.Length<numRows ||numRows==1)
            return s;
        string[] strArray=new string[numRows];
        for(int i =0;i<s.Length;i++){
            int k=i%(2*numRows-2);
            if(k<numRows)
                strArray[k]+=s[i];
            else
                strArray[2*numRows-k-2]+=s[i];
        }
        string rep="";
        foreach(var str in strArray){
            rep+=str;
        }
        return rep;
    }
}
